praw/models/reddit/flair.py

```diff
@@ -0,0 +1,256 @@
+"""Provide the Flair class."""
+from typing import Any, Dict, Optional, TypeVar, Union
+from .base import RedditBase
+
+_AdvancedSubmissionFlair = TypeVar("_AdvancedSubmissionFlair")
+_LinkFlair = TypeVar("_LinkFlair")
+_RedditorFlair = TypeVar("_RedditorFlair")
+_RichFlairBase = TypeVar("_RichFlairBase")
+Reddit = TypeVar("Reddit")
+Submission = TypeVar("Submission")
+Subreddit = TypeVar("Subreddit")
+
+
+class RichFlairBase(RedditBase):
```

Comment:
Is this ever used directly?

Reply:
Not really, but the two subclassess have shared methods, which I have removed for this PR.